spring-integration-core/src/main/java/org/springframework/integration/handler/LoggingHandler.java

```diff
@@ -85,8 +91,7 @@ public LoggingHandler(String level) {
 	 * @since 4.3
 	 */
 	public LoggingHandler(Level level) {
-		Assert.notNull(level, "'level' cannot be null");
-		this.level = level;
+		setLevel(level);
```

Comment:
OK, @englishman , everything looks good. 
Just several concerns, which might not be critical, but would be better to follow.
Just because that is project style.

You can't call non-`final` methods from ctors: http://stackoverflow.com/questions/7477553/in-java-is-there-a-legitimate-reason-to-call-a-non-final-method-from-a-class-co
And quoting:

> In general it's a bad idea to call a non-final method within a constructor for precisely this reason - the subclass constructor body won't have been executed yet, so you're effectively calling a method in an environment which hasn't been fully initialized.

Since we don't know how wrong `LoggingHandler` is used in the target apps, we can make `setLevel()` `final`, especially for the maintain version `4.3.x`.
Although I guess we can make a `private void doSetLevel()` to overcome that issue :smile: